src/store/root-state.epics.ts

```diff
@@ -21,14 +22,22 @@ const resetKeychainOnInstallEpic: Epic = (action$: Observable<Action>, state$: O
     catchError(err => of(resetKeychainOnInstallAction.fail(err.message)))
   );
 
-const resetApplicationEpic = (action$: Observable<Action>) => {
+const resetApplicationAtSecretsEpic = (action$: Observable<Action>) => {
   return action$.pipe(
     ofType(resetApplicationAction.submit),
     switchMap(() => forkJoin([resetKeychain$(), resetBeacon$()])),
-    map(() => {
+    map(() => resetApplicationAction.success()),
+    catchError(err => of(resetApplicationAction.fail(err.message)))
+  );
+};
+
+const resetApplicationAtStateEpic = (action$: Observable<Action>) => {
```

Comment:
the epic name should be related on the action name, that it handles.  
so, this epic should be renamed to `resetApplicationSuccessEpic` and epic name above should not be changed